src/js/legend.js

```diff
@@ -0,0 +1,198 @@
+            row: ".gpii-ca-pieChart-legend-table-row",
+            legendColorCell: ".gpii-ca-pieChart-legend-table-row-legend-color-cell",
+            labelCell: ".gpii-ca-pieChart-legend-table-row-legend-label-cell",
+            valueCell: ".gpii-ca-pieChart-legend-table-row-legend-value-cell"
+        },
+        events: {
+            onLegendCreated: null  // Fire when the legend is created. Ready to register D3 DOM event listeners
+        },
+        listeners: {
+            "onCreate.create": {
+                funcName: "gpii.chartAuthoring.pieChart.legend.create",
+                args: ["{that}"]
+            }
+        },
+        modelListeners: {
+            dataSet: {
+                funcName: "{that}.draw",
+                excludeSource: "init"
+            }
+        },
+        invokers: {
+            draw: {
+                funcName: "gpii.chartAuthoring.pieChart.legend.draw",
+                args: ["{that}"]
+            }
+        }
+    });
+
+    // Takes the dataSet array and the color array, and returns a consolidated object array to ease sorting and other operations while keeping colors "correct"
+
+    gpii.chartAuthoring.pieChart.legend.consolidateDataAndColors = function(dataSet, colors) {
+        var c = [];
+        for(var i=0; i<dataSet.length; i++) {
+            var d = {
+                id: dataSet[i].id,
+                label: dataSet[i].label,
+                value: dataSet[i].value,
+                color: colors(i)
+            };
+            c.push(d);
+        }
+        return c;
+    };
+
+    gpii.chartAuthoring.pieChart.legend.draw = function(that) {
+
+        var table = that.table,
+            l = that.options.legendOptions,
+            dataSet = gpii.chartAuthoring.pieChart.legend.consolidateDataAndColors(that.model.dataSet,that.color),
+            rowClass = that.classes.row,
+            legendColorCellClass = that.classes.legendColorCell,
+            legendLabelCellClass = that.classes.labelCell,
+            legendValueCellClass = that.classes.valueCell,
+            sort = l.sort;
+        var tbody = table.selectAll("tbody");
+
+        var rows = tbody.selectAll("tr")
+            .data(dataSet, function(d) {
+                return d.id;
+            });
+
+        var addedRows = rows.enter().append("tr");
+
+
+        addedRows
+        .attr({
+            "class": rowClass
+        });
+
+        addedRows
+        .append("td")
+        .attr({
+            "style": function(d, i) {
+                return "background-color: "+d.color+";";
+            },
+            "class": legendColorCellClass
+        });
+
+        addedRows
+        .append("td")
+        .attr({
+            "class": legendLabelCellClass
+        })
+        .html(function(d, i) {
+            return d.label;
+        });
+
+        addedRows.append("td")
+        .attr({
+            "class": legendValueCellClass
+        })
+        .html(function(d,i) {
```

Comment:
The infusion coding convention needs a space in btw the keyword "function" and the open bracket, such as "function (".